azure-pipelines.yml

```diff
@@ -164,77 +112,79 @@ jobs:
 - template: /eng/jobs/bash-build.yml
   parameters:
     additionalMSBuildArgs: /p:OutputRid=rhel.6-x64
-    displayName: Build_Linux_x64_Rhel6
+    displayName: Linux_x64_Rhel6
     dockerImage: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-6-376e1a3-20174311014331
     portableBuild: false
     targetArchitecture: x64
 
-  ################################################################################
-  # Build Build_OSX leg
-  ################################################################################
+################################################################################
+# Build OSX (macOS) leg
+################################################################################
+
 - template: /eng/jobs/osx-build.yml
+  parameters:
+    displayName: OSX
 
-  ################################################################################
-  # Build Windows legs
-  ################################################################################
+################################################################################
+# Build Windows legs
+################################################################################
 
 # Windows Arm
 - template: /eng/jobs/windows-build.yml
   parameters:
-    displayName: Build_Windows_Arm
+    displayName: Windows_Arm
     skipTests: true
     targetArchitecture: arm
 
 # Windows Arm64
 - template: /eng/jobs/windows-build.yml
   parameters:
-    displayName: Build_Windows_Arm64
+    displayName: Windows_Arm64
     skipTests: true
     targetArchitecture: arm64
 
 # Windows x64
 - template: /eng/jobs/windows-build.yml
   parameters:
     buildFullPlatformManifest: true
-    displayName: Build_Windows_x64
+    displayName: Windows_x64
     publishRidAgnosticPackages: true
     targetArchitecture: x64
 
 # Windows x86
 - template: /eng/jobs/windows-build.yml
   parameters:
     buildFullPlatformManifest: true
-    displayName: Build_Windows_x86
+    displayName: Windows_x86
     targetArchitecture: x86
 
+################################################################################
+# Publish official build
```

Comment:
Hmm, right now each job publishes to its own artifact, so the final publish job uses the list of dependencies to download each job's artifacts. But maybe all jobs could upload to a single artifact, instead.

I considered using a single artifact while developing this, but went with multiple because it seemed like the simplest way to distinguish job output. (Jobs produce some of the same filenames and the finalize job needs to pick one.) I didn't have stages in mind, so I didn't look at workarounds--I could probably copy the artifacts to a dir with the job name in it and upload that instead. I'll try it out, I agree this would be great.

Reply:
I pushed a change to implement this. I kicked off a mock build to try it out (https://dev.azure.com/dnceng/internal/_build/results?buildId=255796&view=logs), but since the agent pools are dead right now, I figured I'd push it early if you have time to take a look.

Reply:
The mock build hit something that appears to be flakiness, filed https://github.com/dotnet/core-setup/issues/7128 for followup.